core/services/workflows/metering/metering.go

```diff
@@ -111,24 +112,64 @@ type Report struct {
 	lggr    logger.Logger
 
 	// internal state
+	ready bool
 	mu    sync.RWMutex
-	steps map[ReportStepRef]ReportStep
+	steps map[string]ReportStep
 }
 
 func NewReport(accountID, workflowID, workflowExecutionID string, lggr logger.Logger) *Report {
-	logger := lggr.Named("Metering")
-
+	sugaredLggr := logger.Sugared(lggr).Named("WorkflowEngine").With("workflowID", workflowID, "workflowExecutionID", workflowExecutionID)
```

Comment:
I tend to think it's cleaner to not do an assignment here and instead do the assignment on the `Report.lggr` property directly. 

Also you can do the following to not use the sugared logger at all:
```
logger.With(logger.Named(lggr, "WorkflowEngine"), "workflowID", workflowID, "workflowExecutionID", workflowExecutionID)
```

Reply:
My bad. I missed that you were passing it to both. Makes sense.

I'm not sure what you mean by needing to pass in the engine's sugared logger. You should still be able to use `logger.Logger`.